src/graphql/__tests__/allowNotification.js

```diff
@@ -0,0 +1,115 @@
+jest.mock('../lineClient');
+
+import Client from 'src/database/mongoClient';
+import UserSettings from 'src/database/models/userSettings';
+import { gql } from '../testUtils';
+
+it('context rejects anonymous users', async () => {
+  const result = await gql`
+    mutation UpdateNotification($checked: Boolean!) {
+      allowNotification(allow: $checked) {
+        allowNewReplyUpdate
+        newReplyNotifyToken
+      }
+    }
+  `({
+    checked: true,
+  });
+  expect(result).toMatchInlineSnapshot(`
+    Object {
+      "data": Object {
+        "allowNotification": null,
+      },
+      "errors": Array [
+        [GraphQLError: Invalid authentication header],
+      ],
+    }
+  `);
+});
+
+describe('finds', () => {
+  beforeAll(async () => {
+    if (await UserSettings.collectionExists()) {
+      await (await UserSettings.client).drop();
+    }
+  });
+
+  afterAll(async () => {
+    await (await Client.getInstance()).close();
+  });
+
+  it('creates user setting and set allowNewReplyUpdate to true', () =>
+    expect(
+      gql`
+        mutation UpdateNotification($checked: Boolean!) {
+          allowNotification(allow: $checked) {
+            userId
+            allowNewReplyUpdate
+            newReplyNotifyToken
+          }
+        }
+      `(
+        {
+          checked: true,
+        },
+        {
+          userId: 'user1',
+          userContext: {
```

Comment:
Let's remove user context from unit test to test the case when user sets notification without searching anything.

`__tests__/setting.js`'s unit test should also apply.